File: iosApp/iosApp/Presentation/Cafe/CafeAction.swift
//
//  CafeAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation

enum CafeAction {
    case backTapped
    case changeTab(CafeUiState.TabType)
    case maidTapped(id: String)
    case favoriteTapped
    case writeReviewTapped
    case loadMoreCasts
    case loadMoreNotices
    case loadMoreReviews
    case refresh
    case pagingTriggerDisappeared(CafeUiState.TabType)
    case consumeScrollToTopOnReturn
    case editReview(reviewId: String)
    case deleteReview(reviewId: String)
    case reviewImageTapped(imageUrl: String)
    case reportReview(reviewId: String)
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/cafe/CafeAction.kt
package com.hhp227.concafe.presentation.cafe

sealed interface CafeAction {
    data object ClickBack : CafeAction

    data class ChangeTab(val tab: CafeUiState.TabType) : CafeAction

    data class ClickMaid(val id: String) : CafeAction

    data object ClickFavorite : CafeAction

    data object ClickWriteReview : CafeAction

    data object LoadMoreCasts : CafeAction

    data object LoadMoreNotices : CafeAction

    data object LoadMoreReviews : CafeAction

    data object Refresh : CafeAction

    data object ConsumeScrollToTopOnReturn : CafeAction

    data class EditReview(val reviewId: String) : CafeAction

    data class DeleteReview(val reviewId: String) : CafeAction

    data class ClickReviewImage(val imageUrl: String) : CafeAction

    data class ReportReview(val reviewId: String) : CafeAction
}
